servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -743,7 +743,7 @@ void markExpired() {
         }
 
         void markHealthy(final HealthCheck<Addr, C> originalHealthCheckState) {
-            // Marking healthy is generally called from a successful health check, after a connection was added.
+            // Marking healthy is called when we need to recover from an unexpected error.
             // However, it is possible that in the meantime, the host entered an EXPIRED state, then ACTIVE, then failed
             // to open connections and entered the UNHEALTHY state before the original thread continues execution here.
             // In such case, the flipped state is not the same as the one that just succeeded to open a connection.
```

Comment:
(line 753) what if `previous.state == CLOSED`? Should we not set back to `ACTIVE`?

Reply:
It flips only if `Host.isUnhealthy(previous)`